packages/@aws-cdk/aws-sns/lib/subscriber.ts

```diff
@@ -24,6 +24,15 @@ export interface TopicSubscriptionConfig extends SubscriptionOptions {
    * subscribing to.
    */
   readonly subscriberId: string;
+
+  /**
+   * The resources that need to be created before the subscription can be safely created.
+   * For example for SQS subscription, the subscription needs to have a dependency on the SQS queue policy
+   * in order for the subscription to successfully deliver messages.
+   *
+   * @default - empty list
+   */
+  readonly subscriptionDependencies?: IDependable[];
```

Comment:
You're only adding one dependable here so why make it an array? Also, your logic above doesn't seem to have a scenario where subscriptionDependencies is undefined so I'm not sure why this would be optional.

Reply:
I made it an array to make it generic, but I guess it's not backwards incompatible to change this in the future to a list if needed.

It needs to be undefined because only the SQS subscription needs to set this, the lambda and email subscription don't need this: https://github.com/aws/aws-cdk/blob/v1.159.0/packages/%40aws-cdk/aws-sns-subscriptions/lib/email.ts, https://github.com/aws/aws-cdk/blob/v1.159.0/packages/%40aws-cdk/aws-sns-subscriptions/lib/sms.ts, https://github.com/aws/aws-cdk/blob/v1.159.0/packages/%40aws-cdk/aws-sns-subscriptions/lib/lambda.ts